agent-inject/agent/annotations.go

```diff
@@ -257,6 +284,39 @@ func secrets(annotations map[string]string) []*Secret {
 	return secrets
 }
 
+func getNameFromAnnotation(name string, annotations map[string]string) string {
+	secretNamePrefix := fmt.Sprintf("%s-", AnnotationAgentInjectSecret)
+	locationSecretPrefix := fmt.Sprintf("%s-", AnnotationAgentInjectLocation)
+	if strings.Contains(name, secretNamePrefix) {
+		raw := strings.ReplaceAll(name, secretNamePrefix, "")
+		name = strings.ToLower(raw) // why would we force-lowercase the filename here?
+		if len(name) >= 21 || strings.Contains(name, "/") {
+			//houston we have a problem, skip secret, can we log this somewhere?
```

Comment:
👍

Reply:
We should not skip but return a failure response IMHO. Errors should never pass silently. Container coming up with missing env var can fail or might run on a default that might cause a security issue.

Reply:
Yes we can use the error class here.